packages/atomic/src/components/result-lists/atomic-result-children/atomic-result-children.tsx

```diff
@@ -140,31 +162,91 @@ export class AtomicResultChildren {
     );
   }
 
-  public render() {
-    if (!this.ready) return null;
-    if (this.templateHasError) return <slot></slot>;
+  private showCollapseButton() {
     const result = this.getResult();
-    if (result.children.length) {
-      return (
-        <Host>
-          <slot name="before-children"></slot>
-          {result.isLoadingMoreResults ? (
-            <Loading></Loading>
-          ) : (
-            result.children.map((child) => this.renderChild(child))
-          )}
-          <slot name="after-children"></slot>
-        </Host>
-      );
+    return (
+      Boolean(result.children.length) &&
+      !result.isLoadingMoreResults &&
+      !result.moreResultsAvailable
+    );
+  }
+
+  private showChildrenWrapper() {
```

Comment:
The name of this function makes me instinctively think it's an action rather than a condition/getter.

I would either make this a getter (`private get showChildrenWrapper()`) or prefix it with a verb like `should`

Reply:
yeah, I agree. could you explain why making it a getter would make it better? and also in general when it is that you decide to use getters?